cli/app.go

```diff
@@ -224,6 +231,49 @@ var dataTagByFilterFlags = append([]cli.Flag{
 },
 	commonFilterFlags...)
 
+func getValFromContext(name string, ctx *cli.Context) interface{} {
+	// some fuzzy searching is required here, because flags are typically in kebab case, but
+	// params are typically in snake or camel case
+	replacer := strings.NewReplacer("_", "-")
+	dashFormattedName := replacer.Replace(strings.ToLower(name))
+
+	value := ctx.Value(dashFormattedName)
+	if value != nil {
+		return value
+	}
+
+	camelFormattedName := matchFirstCap.ReplaceAllString(name, "${1}-${2}")
+	camelFormattedName = matchAllCap.ReplaceAllString(camelFormattedName, "${1}-${2}")
+	camelFormattedName = strings.ToLower(camelFormattedName)
+
+	return ctx.Value(camelFormattedName)
+}
+
+func createCommandWithT[T interface{}](f func(T, *cli.Context) error) cli.ActionFunc {
+	return func(ctx *cli.Context) error {
+		var t T
+		tValue := reflect.ValueOf(&t).Elem()
+		tType := tValue.Type()
+		for i := 0; i < tType.NumField(); i++ {
+			field := tType.Field(i)
+			value := getValFromContext(field.Name, ctx)
+			tValue.Field(i).Set(reflect.ValueOf(value))
+
+		}
+		return f(t, ctx)
+	}
+}
+
+type foo struct {
+	FooFoo string
+	Bar    int
+}
+
+func doFoo(foo foo, ctx *cli.Context) error {
+	fmt.Printf("FooFoo is %s and Bar is %v.", foo.FooFoo, foo.Bar)
+	return nil
+}
```

Comment:
This will get deleted in the final project obviously, but I wanted to show an example of what new development would look like. Also, it highlights how we successfully set a field `FooFoo` with a flag of `foo-foo`